### PR TITLE
Restricted Genetics and possibly other Alpha traits

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Content "Restricted Genetics trait for Alpha"

#### Purpose of change
Alpha focuses on the superiority of it's genome, mentally and as a category.  This will expand that facet of the tree.

#### Describe the solution
Adds a trait that restricts the PC from getting new traits outside of Alpha.

#### Describe alternatives you've considered
Adding two more mutations as well. 

#### Testing
Tested and appears to work

### CODE REVIEW

**Comment by Hyperseeker** [comment]
The fact that mutations make evolution viable makes me think that Alpha is a doomed genetic line. Without mutations (or, I guess, extra mutations), Alpha mutants are doomed to fail because they're inflexible. Not that the game is modelling life on an evolutionary scale, but... it makes me wonder if this is something worth emphasizing even in the short term. Yes, Alphas are *superior*, for a given sense of the word. Are they not, then, a prime example of hubris, and are thus forging their own path to failure?

I know they get some of the mental traits that make them *think* they're superior to everybody else. I wonder if at the higher levels they could also get... I dunno, a trait that prevents them from forming antibodies against non-common diseases? Or maybe prevent their skills from rising? Something about "I am perfect as I am, I cannot possibly be more".

**Comment by fairyarmadillo** [comment]
A minor suggestion - it sounded like Alpha needs more mutations, a simple precursor trait for this one could be one that does the inverse of what robust genetics does.

**Comment by Maleclypse** [comment]
> A minor suggestion - it sounded like Alpha needs more unique mutations, a simple precursor trait for this one could be one that does the inverse of what robust genetics does.

I’m trading some json work for the c++ on the precursor trait. Precursor trait will multiply vitamin costs for other mutation categories.

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -238,6 +238,7 @@
       "RESISTCHILL_RESISTWARM",
       "RESISTWARM",
       "ROBUST",
+      "RESTRICTED",
```
Comment (Maleclypse) [inline data/json/mutations/mutations.json:241]:
```suggestion
      "RESTRICTED"
```

File: data/json/mutations/mutations.json [hunk 2]
```diff
@@ -238,6 +238,7 @@
       "RESISTCHILL_RESISTWARM",
       "RESISTWARM",
       "ROBUST",
+      "RESTRICTED"
```
Comment (Maleclypse) [inline data/json/mutations/mutations.json:241]:
```suggestion
      "RESTRICTED"
```

File: data/json/mutations/mutations.json [hunk 3]
```diff
@@ -1205,6 +1206,16 @@
     "cancels": [ "CHAOTIC_BAD" ],
     "category": [ "FISH", "SLIME", "MEDICAL", "PLANT" ]
   },
+  {
+    "type": "mutation",
+    "id": "RESTRICTED",
+    "name": { "str": "Restricted Genetics" },
+    "points": 3,
```
Comment (Maleclypse) [inline data/json/mutations/mutations.json:1213]:
```suggestion
    "points": 3,
    "vitamin_cost": 160,
```

**Pull request merged**